aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1720,21 +1732,37 @@ static Tensor& _linalg_norm_vector_out(Tensor& result, const Tensor& self, optio
     TORCH_CHECK(unique_dims, "Expected dims to be different, got this instead: (", dim, ")");
   }
   if (!case_was_overridden) {
-    if (opt_dtype.has_value()) {
-      result_ = at::norm(self.to(opt_dtype.value()), opt_ord, dim, keepdim);
+    if (flatten) {
+      if (opt_dtype.has_value()) {
+        result_ = at::norm(self.to(opt_dtype.value()), opt_ord.value_or(2));
+      } else {
+        result_ = at::norm(self, opt_ord.value_or(2));
+      }
+      if (keepdim) {
+        result_ = result_.reshape(std::vector<int64_t>(self.dim(), 1));
+      }
     } else {
-      result_ = at::norm(self, opt_ord, dim, keepdim);
+      if (opt_dtype.has_value()) {
+        result_ = at::norm(self.to(opt_dtype.value()), opt_ord, dim, keepdim);
+      } else {
+        result_ = at::norm(self, opt_ord, dim, keepdim);
+      }
     }
   }
   resize_output(result, result_.sizes());
   result.copy_(result_);
```

Comment:
And this, as you've discussed, applies to all cases, not just newly added `flatten`. So yeah, this really needs to be fixed. How come previous performance tests did not catch this?